syntax = "proto3";

package srv6_explicit_path;

// Define the rpc service interface
service SRv6ExplicitPath {
  // Create operation
  rpc Create (SRv6EPRequest) returns (SRv6EPReply) {}
  // Remove operation
  rpc Remove (SRv6EPRequest) returns (SRv6EPReply) {}
  rpc Replace (SRv6EPRequest) returns (SRv6EPReply) {}
}

// The SRv6EPRequest message containing a number of paths.
message SRv6EPRequest {
  string src = 1;
  string dst = 2;
  message SRv6Segment {
    string segment = 1;
  }
  repeated SRv6Segment sr_path = 3;
  string encapmode = 4;
  string device = 5;
}



// The SRv6EPReply message containing a simple status message
message SRv6EPReply {
  string message = 1;
}
